packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts

```diff
@@ -1,89 +1,94 @@
+import { RootXStoreState } from '../../../../store';
+import { XPlugin } from '../../../../plugins';
+import { UrlParams } from '../../../../types';
 import { QueryPreviewInfo, QueryPreviewItem } from '../../store/types';
 import { queriesPreviewXModule } from '../../x-module';
 import QueryPreview from '../query-preview.vue';
-import { getEmptySearchResponseStub } from '../../../../__stubs__/index';
 import { getHashFromQueryPreviewInfo } from '../../utils/get-hash-from-query-preview';
 import { resetXQueriesPreviewStateWith } from './utils';
 
-function renderQueryPreview({
-  maxItemsToRender,
-  queryPreviewInfo = { query: 'milk' },
-  location,
-  queryFeature,
-  persistInCache = false,
+async function render({
+  template = `<QueryPreview :queryPreviewInfo="queryPreviewInfo" :queryFeature="queryFeature" :maxItemsToRender="maxItemsToRender" :debounceTimeMs="debounceTimeMs" :persistInCache="persistInCache"/>`,
+  queryPreviewInfo = { query: 'milk' } as QueryPreviewInfo,
+  queryFeature = undefined as undefined | string,
+  maxItemsToRender = undefined as undefined | number,
   debounceTimeMs = 0,
-  template = `<QueryPreview v-bind="$attrs" />`,
+  persistInCache = false,
+  location = undefined as undefined | string,
   queryPreviewInState = {
-    request: {
-      query: queryPreviewInfo.query
-    },
+    request: {},
     results: getResultsStub(4),
     status: 'success',
     instances: 1,
     totalResults: 100
-  }
-}: RenderQueryPreviewOptions = {}): RenderQueryPreviewAPI {
+  } as QueryPreviewItem | null
+} = {}) {
   const store = new Store<DeepPartial<RootXStoreState>>({});
-  XPlugin.registerXModule(queriesPreviewXModule);
-
-  const queryPreviewRequestUpdatedSpy = jest.fn();
-  XPlugin.bus.on('QueryPreviewRequestUpdated').subscribe(queryPreviewRequestUpdatedSpy);
-
-  const queryPreviewUnmounted = jest.fn();
-  XPlugin.bus.on('QueryPreviewUnmounted').subscribe(queryPreviewUnmounted);
-
-  if (queryPreviewInState) {
-    resetXQueriesPreviewStateWith(store, {
-      queriesPreview: {
-        [getHashFromQueryPreviewInfo(queryPreviewInfo)]: queryPreviewInState
-      }
-    });
-  }
 
-  const wrapper: VueWrapper = mount(
+  const wrapper = mount(
     {
+      template,
       components: { QueryPreview },
-      template
+      props: [
+        'queryPreviewInfo',
+        'queryFeature',
+        'maxItemsToRender',
+        'debounceTimeMs',
+        'persistInCache'
+      ]
     },
     {
       global: {
         plugins: [installNewXPlugin({ store, initialXModules: [queriesPreviewXModule] })],
-        provide: {
-          location
-        }
+        provide: { location }
       },
-      store,
       props: {
-        maxItemsToRender,
         queryPreviewInfo,
         queryFeature,
+        maxItemsToRender,
         debounceTimeMs,
         persistInCache
       }
     }
-  ).findComponent(QueryPreview);
+  );
+
+  if (queryPreviewInState) {
```

Comment:
Pretending like I didn't see an `if` in a test 🫣

Reply:
😅 I could remove it, making `queryPreviewInState` param required and passing the state when it is needed